config.proto

```diff
@@ -127,3 +130,17 @@ message JavaAgent {
     // filter_jar_paths is the list of path to filter jars, separated by `,`
     repeated google.protobuf.StringValue filter_jar_paths = 1;
 }
+
+// CustomDataCaptureEndpoint represents a custom data capture rule for an endpoint that matches
+// the given url pattern and host header
+message CustomDataCaptureEndpoint {
+
+    // data_capture describes the data being captured by instrumentation for this endpoint
+    DataCapture data_capture = 1;
+
+    // url_pattern describes the url pattern to match for this endpoint
+    google.protobuf.StringValue url_pattern = 2;
```

Comment:
We should move `path_pattern` and `host_header` to a separate message. Something like `Match` (there must be a better name) which can be `oneof` different ways of matching against a span. One of them being UrlPathMatch which contains these two. This way the match behavior is more extensible.

Reply:
I'd rather not do that. What identify the capture is the matching criteria, so having same settings for many matching criteria sounds more like a microoptimization. I'd keep it as it is.